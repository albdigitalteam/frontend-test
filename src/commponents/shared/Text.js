import styled, { css } from 'styled-components';

const Text = styled.Text`
  color: ${props => props.color || '#000000'};
  font-size: ${props => props.size ? props.size : 16}px;
  line-height: 24px;
  text-align: ${props => props.align};
  font-weight: ${props => props.weight};
  ${props => props.small && css`
    font-size: 10px;
    line-height: 16px;
  `}
  ${props => props.caption && css`
    font-size: 12px;
    line-height: 16px;
  `}
  ${props => props.helper && css`
    font-size: 14px;
    line-height: 20px;
  `}
  ${props => props.title && css`
    font-size: 20px;
    line-height: 32px;
  `}
  ${props => props.headline && css`
    font-size: 24px;
    line-height: 36px;
  `}
  ${props => props.link && css`
    text-decoration-line: underline;
  `}
`

Text.defaultProps = {
  align: 'left',
  weight: 'normal',
  theme: {
    text: '#000000'
  }
}

export default Text
