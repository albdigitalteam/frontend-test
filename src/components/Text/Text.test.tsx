import React from 'react';
import { render } from '@testing-library/react-native';
import Text from './Text';

describe('test text component', () => {
  it('render and verify text content', () => {
    const testText = 'Testing text';
    const { getByText } = render(<Text>{testText}</Text>);

    expect(getByText(testText)).toBeTruthy();
  });
});

describe('snapshopt test text component', () => {
  it('render text component', () => {
    const testText = 'Testing text';
    const tree = render(<Text>{testText}</Text>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
