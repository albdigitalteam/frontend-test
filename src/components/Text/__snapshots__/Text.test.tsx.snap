// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshopt test text component render text component 1`] = `
<Text
  style={
    Array [
      Object {
        "display": "flex",
      },
    ]
  }
>
  Testing text
</Text>
`;
